lib/from_openstudio/model.rb

```diff
@@ -178,5 +201,49 @@ def self.materials_from_model(openstudio_model)
       result
     end
 
+    # Create HB Construction from OpenStudio Materials
+    def self.constructions_from_model(openstudio_model)
+      result = []
+
+      # Create HB AirConstruction from OpenStudio Construction
+      openstudio_model.getConstructionAirBoundarys.each do |construction|
+        result << AirBoundaryConstructionAbridged.from_construction(construction)
+      end
+
+      # Create HB WindowConstruction from OpenStudio Construction
+      openstudio_model.getConstructions.each do |construction|
+        if construction.isFenestration
+          window_construction = false
+          construction.layers.each do |material|
+            if material.to_StandardGlazing.is_initialized or material.to_SimpleGlazing.is_initialized or material.to_Gas.is_initialized or material.to_GasMixture.is_initialized
+              window_construction = true
+            end
+          end
+          if window_construction == true
+            result << WindowConstructionAbridged.from_construction(construction)
+          end
+        else
+          opaque_construction = false
+          construction.layers.each do |material|
```

Comment:
Again, you don't need to loop through all of these. Just grab the first layer. Also, I think you'll need to do a check for NoMass materials as well as Standard Opaque Materials.  You don't need to check for AirGap materials if you only check the first/outer-most layer since it's illegal to have them on the outside of a Construction in E+.